### PR TITLE
[PDI-17667] Use of vulnerable component jetty 8.1.15, CVE-2017-9735, …

### PR DESCRIPTION
**Pull request by smmribeiro**
…CVE-2017-7657 CVE-2015-2080

[PPP-4183] Use of Vulnerable Component: Multiple Jetty Components [Listed in Description] (CVE-2017-7656 | CVE-2017-7657 | CVE-2017-7658 | CVE-2015-2080 | CVE-2017-9735 )

Centralized Jetty components: use dependency management.
On the 'assembly.xml':
- Added 'org.eclipse.jetty:jetty-xml' and fixed 'pentaho:pentaho-metadata' to 'org.pentaho:pentaho-metadata' - they're pom dependencies (and already exist in the final installation folder)
- Removed the duplicated entries for 'xmlpull:xmlpull' and 'xpp3:xpp3_min'
- Removed 'org.codehaus.jackson:jackson-core-asl' and 'pentaho:pentaho-xul-core' - they weren't being included on the resulting Zip file

Depends on pentaho/maven-parent-poms#121

@pentaho-lmartins @pentaho/tatooine
@graimundo @nantunes

### CODE REVIEW

File: assemblies/pmr-libraries/src/main/descriptors/assembly.xml [hunk 1]
```diff
@@ -158,6 +158,7 @@
         <include>org.eclipse.jetty:jetty-server</include>
         <include>org.eclipse.jetty:jetty-servlet</include>
         <include>org.eclipse.jetty:jetty-util</include>
+        <include>org.eclipse.jetty:jetty-xml</include>
```
Comment (graimundo) [inline assemblies/pmr-libraries/src/main/descriptors/assembly.xml]: @smmribeiro did you double check with the @pentaho/moseisley team if this is needed to be included?

Comment (smmribeiro) [reply assemblies/pmr-libraries/src/main/descriptors/assembly.xml]: No, I did not, @graimundo.

There’re 9 jetty components’ dependencies declared (<big-data-plugin>\assemblies\pmr-libraries\pom.xml) and of those, only jetty-xml is not included in the assembly; **however**, all 9 exist on the corresponding lib folder on an installation (<PENTAHO_FOLDER>\design-tools\data-integration\lib).
As so, I’ve added the component to the assembly.xml
Did I assume wrongly? Do you think I should check with them?

On a side note: there're a few other inconsistencies between pom.xml and assembly.xml (3 dependencies that are not assembled, and 10 jars that are assembled and aren't dependencies).

Comment (graimundo) [reply assemblies/pmr-libraries/src/main/descriptors/assembly.xml]: > Did I assume wrongly? Do you think I should check with them?

I don't know if you assumed wrongly, hence we should ask @pentaho/moseisley as they have the domain knowledge here. Please double check with them. Also bring up the remaining inconsistencies as we might take the chance and clean those up as well.

Comment (smmribeiro) [reply assemblies/pmr-libraries/src/main/descriptors/assembly.xml]: Resolving as I'm talking to @pentaho/moseisley.

**Comment by smmribeiro** [comment]
Closing PR: to simplify Karaf's upgrade a different approach was defined!

**Pull request closed without merging**